calculator_test.go

```diff
@@ -2,23 +2,212 @@ package calculator_test
 
 import (
 	"calculator"
+	"math/rand"
 	"testing"
 )
 
 func TestAdd(t *testing.T) {
+	testcases := []struct {
+		a, b float64
+		rest []float64
+		want float64
+		name string
+	}{
+		{name: "Add two units", a: 0, b: 0, rest: []float64{}, want: 0},
+		{name: "Add two nonunits", a: 1, b: 2, rest: []float64{}, want: 3},
+		{name: "Add unit and nonunit", a: 0, b: 2, rest: []float64{}, want: 2},
+		{name: "Add negative and inverse", a: -2, b: 2, rest: []float64{}, want: 0},
+		{name: "Add negative and positive", a: -2, b: 3, rest: []float64{}, want: 1},
+		{name: "Add two numbers", a: 1, b: 2, rest: []float64{}, want: 3},
+		{name: "Add three numbers", a: 1, b: 2, rest: []float64{3}, want: 6},
+		{name: "Add four numbers", a: 1, b: 2, rest: []float64{3, 4}, want: 10},
+		{name: "Add five numbers", a: -1, b: 2, rest: []float64{-3, 4, -5}, want: -3},
+	}
+	for _, testcase := range testcases {
```

Comment:
Try `tc` here; we're going to use this name a lot, and it will be laborious both to write and to read `testcase` every time.